src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/ConstantExpectedAnalyzer.cs

```diff
@@ -0,0 +1,566 @@
+                case SpecialType.System_Double:
+                    return UnmanagedHelper<double>.Validate(parameterSymbol, attributeData, double.MinValue, double.MaxValue, Helper, out diagnostics);
+                case SpecialType.System_Boolean:
+                    return UnmanagedHelper<bool>.Validate(parameterSymbol, attributeData, false, true, Helper, out diagnostics);
+                case SpecialType.System_String:
+                    return Validate(parameterSymbol, attributeData, Helper, out diagnostics);
+                case SpecialType.None when parameterSymbol.Type.TypeKind == TypeKind.TypeParameter:
+                    return Validate(parameterSymbol, attributeData, Helper, out diagnostics);
+                default:
+                    diagnostics = Helper.ParameterIsInvalid(parameterSymbol.Type.ToDisplayString(), attributeData.ApplicationSyntaxReference.GetSyntax());
+                    return false;
+            }
+        }
+
+        private static bool TryGetConstantExpectedAttributeData(IParameterSymbol parameter, [NotNullWhen(true)] out AttributeData? attributeData)
+        {
+            AttributeData? constantExpectedAttributeData = parameter.GetAttributes()
+                .FirstOrDefault(attrData => IsConstantExpectedAttribute(attrData.AttributeClass));
+            attributeData = constantExpectedAttributeData;
+            return constantExpectedAttributeData is not null;
+        }
+
+        private static bool HasConstantExpectedAttributeData(IParameterSymbol parameter)
+        {
+            return parameter.GetAttributes()
+                .Any(attrData => IsConstantExpectedAttribute(attrData.AttributeClass));
+        }
+
+        private static bool IsConstantExpectedAttribute(INamedTypeSymbol namedType)
+        {
+            return namedType.Name.Equals(ConstantExpectedAttribute, StringComparison.Ordinal) &&
+                   namedType.GetMembers().OfType<IPropertySymbol>()
+                       .All(s => s.Name.Equals("Min", StringComparison.Ordinal) ||
+                                 s.Name.Equals("Max", StringComparison.Ordinal));
+        }
+
+        private abstract class ConstantExpectedParameter
+        {
+            protected ConstantExpectedParameter(IParameterSymbol parameter)
+            {
+                Parameter = parameter;
+            }
+
+            public IParameterSymbol Parameter { get; }
+
+            public abstract bool ValidateValue(IArgumentOperation argument, object? constant, [NotNullWhen(false)] out Diagnostic? validationDiagnostics);
+            public abstract bool ValidateParameterIsWithinRange(ConstantExpectedParameter subsetCandidate, IArgumentOperation argument, [NotNullWhen(false)] out Diagnostic? validationDiagnostics);
+        }
+
+        private sealed class StringConstantExpectedParameter : ConstantExpectedParameter
+        {
+            public StringConstantExpectedParameter(IParameterSymbol parameter) : base(parameter) { }
```

Comment:
```suggestion
            public StringConstantExpectedParameter(IParameterSymbol parameter) : base(parameter) {
            }
```
nit

Reply:
hmm, I thought we either do, I know the runtime does the first one for empty methods
```cs
   void A() { }
```
or
```cs
   void A() 
   {
   }
```

Reply:
second one.